chain/src/txhashset/txhashset.rs

```diff
@@ -1323,13 +1315,23 @@ impl<'a> Extension<'a> {
 		let total_kernels = pmmr::n_leaves(self.kernel_pmmr.unpruned_size());
 		for n in 1..self.kernel_pmmr.unpruned_size() + 1 {
```

Comment:
Why not use the new iter here

Reply:
Good question! The kernel MMR is non-prunable so we do not actually maintain a leaf_set (all leaves are always in the leaf_set). And we have not (yet) implemented the new iterator for this case.